### PR TITLE
(SIMP-8699) Re-worked resolv module

### PR DESCRIPTION
**Pull request by trevor-vaughan**
- Add the ability to precisely update the resolv.conf contents
- Add the ability to specify the entire contents of the resolv.conf
- Add the ability to remove the resolv.conf completely

SIMP-8699 #close

### CODE REVIEW

**Review by jeannegreulich — CHANGES_REQUESTED** [review]
When I have two interfaces controlled by Network Manager, it wipes out all the stuff from the interface not noted in hiera for the resolv module.  Should 1) you also be adding the search domains to ipv4.search
and 2) not removing the stuff from the other interface but only updating the options line.

File: manifests/init.pp [hunk 1]
```diff
@@ -1,179 +1,270 @@
...
 
-        # Add the specified nameservers unless they are already configured for the given device
-        exec { 'Add DNS servers via nmcli':
-          command => "nmcli connection modify \"${nmcli_connection_name}\" ipv4.ignore-auto-dns ${nmcli_ignore_auto_dns} ipv4.dns \"${_flattened_name_servers}\"",
-          unless  => "[ \"\$( nmcli -f ipv4.dns connection show \"${nmcli_connection_name}\" | awk '{print \$2}' | tr '\\n' ' ' | tr ',' ' ' )\" == \"${_flattened_name_servers} \" ]",
-          path    => '/bin:/usr/bin',
-        }
+          # Add the specified nameservers unless they are already configured for the given device
+          exec { 'Add DNS servers via nmcli':
+            command => "nmcli connection modify \"${nmcli_connection_name}\" ipv4.ignore-auto-dns ${nmcli_ignore_auto_dns} ipv4.dns \"${_flattened_name_servers}\"",
```
Comment (jeannegreulich) [inline manifests/init.pp:173]: Do you also want to add the search domains to the connection?

Comment (trevor-vaughan) [reply manifests/init.pp:173]: Yeah, it looks like we might want to pass all of the various options into nmcli. It's a shame that it's so clunky.

**Comment by kendall-moore** [comment]
This PR is deprecating a feature that was added 5 months ago in direct response to a production outage from a customer and as far as I can tell is not offering a comparable way to achieve the same functionality (managing resolv via nmcli). I don't think that this should be allowed to be merged until there's an actual solution for nmcli that users can test in the meantime. Once that exists if we want to deprecate managing resolv via nmcli in this module then I think that's totally fine.

**Review by op-ct — APPROVED** [review]
:eye: 👍

**Review by jeannegreulich — APPROVED** [review]

**Pull request merged**